components/console-backend-service/internal/domain/servicecatalog/serviceplan_resolver.go

```diff
@@ -0,0 +1,68 @@
+package servicecatalog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/domain/shared"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlerror"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlschema"
+)
+
+type servicePlanResolver struct {
+	sc     *servicePlanService
+	rafter shared.RafterRetriever
+}
+
+func newServicePlanResolver(sc *servicePlanService, r shared.RafterRetriever) *servicePlanResolver {
+	return &servicePlanResolver{sc, r}
+}
+
+func (r *servicePlanResolver) ServicePlanClusterAssetGroupField(ctx context.Context, obj *gqlschema.ServicePlan) (*gqlschema.ClusterAssetGroup, error) {
+	if obj == nil {
+		glog.Error(fmt.Errorf("while getting clusterAssetGroup field obj is empty"))
+		return nil, gqlerror.NewInternal()
+	}
+	return r.getClusterAssetGroup(obj.Name), nil
+}
+
+func (r *servicePlanResolver) ServicePlanAssetGroupField(ctx context.Context, obj *gqlschema.ServicePlan) (*gqlschema.AssetGroup, error) {
+	if obj == nil {
+		glog.Error(fmt.Errorf("while getting assetGroup field obj is empty"))
+		return nil, gqlerror.NewInternal()
+	}
+	return r.getAssetGroup(obj.Name, obj.Namespace), nil
+}
+
+func (r *servicePlanResolver) getClusterAssetGroup(name string) *gqlschema.ClusterAssetGroup {
```

Comment:
This function also is not needed. You can add whole logic only in one function `ServicePlanClusterAssetGroupField `, not split to two.